### DISCUSSION TITLE
有大佬在arm下编译成功么

### DISCUSSION THREAD

**Discussion by a492486** [kbengine/kbengine#1526] [General]
我有个arm的机子，想把kbengine整上去试试,编译的时候好多地方不认识arm架构，有成功的大佬么

**Comment by a492486** [comment]
1. /kbengine-2.5.10/kbe/src/build/common.mak 
	修改ARCHFLAGS=-m32修改
	为ARCHFLAGS=
2. /kbengine-2.5.10/kbe/src/lib/dependencies/hiredis/Makefile 
	修改REAL_CFLAGS=$(OPTIMIZATION) -fPIC $(CFLAGS) $(WARNINGS) $(DEBUG) $(ARCH)
	REAL_LDFLAGS=$(LDFLAGS) $(ARCH)
	为
	REAL_CFLAGS=$(OPTIMIZATION) -fPIC $(CFLAGS) $(WARNINGS) $(DEBUG) 
	REAL_LDFLAGS=$(LDFLAGS) 

3. /kbengine-2.5.10/kbe/src/lib/dependencies/expat/Makefile
4. /root/kbengine-2.5.10/kbe/src/lib/dependencies/log4cxx/Makefile
	修改CONFIG_ARGS =
	为CONFIG_ARGS =--build=arm-linux


5. /root/kbengine-2.5.10/kbe/src/lib/common/timestamp.h

将“lib/common/timestamp.h”中的timestamp_rdtsc()函数，替换为如下代码：

inline uint64 timestamp_rdtsc()

{
#if defined(__x86_64__) || defined(__amd64__)

uint32 rethi, retlo;

__asm__ __volatile__(

"rdtsc":

"=d"    (rethi),

"=a"    (retlo)                                    );

return uint64(rethi) << 32 | retlo

#elif defined(__aarch64__)

int64_t virtual_timer_value;

asm volatile("mrs %0, cntvct_el0" : "=r"(virtual_timer_value));

return virtual_timer_value;

#else

#error "not support"

#endif

}

6. /root/kbengine-2.5.10/kbe/src/lib/server/telnet_handler.cpp
     case  -1:      修改为      case (char)-1: 


7. apt-get install  libmysqlclient-dev

**Comment by huhubis** [comment]
怎么有这种需求。  记得以前有人成功过
